osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -62,4 +71,43 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
 
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+		for (VirtualSystem vs : da.getVirtualSystems()) {
+			validateVSReferenceToSfcAndSecurityGroup(em, da, vs);
+		}
+	}
+
+	private void validateVSReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da, VirtualSystem vs)
+			throws Exception {
+
+		if (!getSFCs(vs)) {
+			if (validateSfcReferenceToSecurityGroup(em, da, vs) != null) {
```

Comment:
This pattern is very strange. A method or function returning an exception is a code smell and should always be avoided. Exceptions are usually meant to be thrown not returned. Especially here where you check to see if it returns and exception and if so you call the same method again but then you throw. instead simply make the method validateSfcReferenceToSG throw and call it for each sfc in the VS looping through the sfcs . You also should not need to pass the VS as a parameter for this method at all since you would have the loop of ln 93 on this method instead. The pattern is... if empty return, if not loop through the sfcs in the VS, for each SFC call validateSfcRErferencetoSG which does not take a VS and does not return anything.